### PR TITLE
Redirect to LMS login if needed and possible

### PR DESCRIPTION
**Pull request by shadinaif**
## Change description

See https://github.com/appsembler/edx-platform/pull/1301

## Type of change
- [ ] Bug fix (fixes an issue)
- [x] New feature (adds functionality)

## Related issues

Related to: https://github.com/appsembler/edx-platform/pull/1301

### Development

- [ ] Lint rules pass locally
- [ ] Application changes have been tested thoroughly
- [ ] Automated tests covering modified code pass

### Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### Code review 

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

File: cms/templates/login_page.html [hunk 1]
```diff
@@ -6,18 +6,40 @@
...
+<%def name="get_redirect_url_to_lms()"><%
+return tahoe_idp_helpers.get_redirect_to_lms_login_url(get_current_request()) if cusomized_form else ''
+%></%def>
+
 <%block name="title">${_("Sign In")}</%block>
 <%block name="bodyclass">not-signedin view-signin</%block>
 
 <%block name="content">
 
+<span id="redirect_url_span" style="display: none;">${get_redirect_url_to_lms()}</span>
```
Comment (OmarIthawi) [inline cms/templates/login_page.html]:
```suggestion
```

File: cms/templates/login_page.html [hunk 2]
```diff
@@ -6,18 +6,40 @@
...
-    <header>
-      <h1 class="title title-1">${_("Sign In to {studio_name}").format(studio_name=settings.STUDIO_NAME)}</h1>
-    </header>
-    %if tahoe_idp_helpers.is_studio_login_form_overridden():
+    %if cusomized_form and get_redirect_url_to_lms():
+      <header>
+        <h1 class="title title-1">
+          ${_("Redirecting to LMS login...")}
+        </h1>
+      </header>
```
Comment (OmarIthawi) [inline cms/templates/login_page.html:30]: Same work, but with less code.

```suggestion
      </header>
        <script id="tahoe-studio-logout-script" type="application/javascript" data-url="${get_redirect_url_to_lms()}">
            const redirectUrl = document.getElementById('tahoe-studio-logout-script').dataset.url;
            window.location = redirectUrl;
        </script>
```

File: cms/templates/login_page.html [hunk 3]
```diff
@@ -59,4 +81,18 @@ <h1 class="title title-1">${_("Sign In to {studio_name}").format(studio_name=set
...
+        // We're using the dummy span (redirect_url_span) to securly fetch the URL
+        const redirect_url = document.getElementById('redirect_url_span').textContent;
+        if (redirect_url) {
+          window.location.replace(redirect_url);
+        }
+    }
+    window.onload = redirect_to_lms_login;
+  </script>
+%endif
+
```
Comment (OmarIthawi) [inline cms/templates/login_page.html]:
```suggestion
```

**Review by OmarIthawi — DISMISSED** [review]
Thanks for getting in the JavaScript maze. I've made the code more JavaScriptic.

File: cms/templates/login_page.html (hunk 1 above)
Comment (shadinaif) [reply cms/templates/login_page.html]: all done here, thank you @OmarIthawi

**Comment by shadinaif** [comment]
Tested on devstack and it's working great, please take a look @OmarIthawi

**Review by OmarIthawi — APPROVED** [review]
Thanks!!

**Pull request merged**